packages/sdk/contractkit/src/wrappers/GrandaMento.test.ts

```diff
@@ -0,0 +1,158 @@
+      )
+      expect(config.exchangeLimits.get(StableTokenName.cUSD)?.maxExchangeAmount).toEqBigNumber(
+        new BigNumber(newLimitMax)
+      )
+      expect(config.exchangeLimits.get(StableTokenName.cEUR)?.minExchangeAmount).toEqBigNumber(
+        new BigNumber(0)
+      )
+      expect(config.exchangeLimits.get(StableTokenName.cEUR)?.maxExchangeAmount).toEqBigNumber(
+        new BigNumber(0)
+      )
+    })
+
+    it('has new limits', async () => {
+      const limits = await grandaMento.stableTokenExchangeLimits(StableTokenName.cUSD)
+      expect(limits.minExchangeAmount).toEqBigNumber(newLimitMin)
+      expect(limits.maxExchangeAmount).toEqBigNumber(newLimitMax)
+    })
+
+    describe('Has  a proposal', () => {
+      beforeEach(async () => {
+        sellAmount = new BigNumber('100000000')
+        await (
+          await celoToken.increaseAllowance(grandaMento.address, sellAmount)
+        ).sendAndWaitForReceipt()
+
+        await (
+          await grandaMento.createExchangeProposal(StableTokenName.cUSD, sellAmount, true)
+        ).sendAndWaitForReceipt()
+      })
+
+      it('executes', async () => {
+        const activeProposals = await grandaMento.getActiveProposalIds()
+        expect(activeProposals).not.toEqual([])
+
+        let proposal = await grandaMento.getExchangeProposal(activeProposals[0])
+        expect(proposal.exchanger).toEqual(accounts[0])
+        expect(proposal.stableToken).toEqual(stableToken.address)
+        expect(proposal.sellAmount).toEqBigNumber(sellAmount)
+        expect(proposal.buyAmount).toEqBigNumber(new BigNumber('99000000'))
+        expect(proposal.approvalTimestamp).toEqual(new BigNumber(0))
+        expect(proposal.state).toEqual(ExchangeProposalState.Proposed)
+        expect(proposal.sellCelo).toEqual(true)
+
+        await (
+          await grandaMento.approveExchangeProposal(activeProposals[0])
+        ).sendAndWaitForReceipt()
+
+        proposal = await grandaMento.getExchangeProposal(activeProposals[0])
+
+        expect(proposal.state).toEqual(ExchangeProposalState.Approved)
+        await timeTravel(expConfig.vetoPeriodSeconds, web3)
+        await (
+          await grandaMento.executeExchangeProposal(activeProposals[0])
+        ).sendAndWaitForReceipt()
+
+        proposal = await grandaMento.getExchangeProposal(activeProposals[0])
+        expect(proposal.state).toEqual(ExchangeProposalState.Executed)
+      })
+
+      it('cancels proposal', async () => {
+        await (await grandaMento.cancelExchangeProposal(1)).sendAndWaitForReceipt()
+
+        const proposal = await grandaMento.getExchangeProposal('1')
+        expect(proposal.state).toEqual(ExchangeProposalState.Cancelled)
+      })
+    })
+  })
+
+  it('#getConfig', async () => {
+    const config = await grandaMento.getConfig()
+    expect(config.approver).toBe(expConfig.approver) // TODO FIX this tests, for some reason `expConfig.approver` is 0x0000...0 even it's writen on the migrations-override.json
```

Comment:
Need some help here

Reply:
and the approver is correct in the migrations, because it can approve in the tests above with `account[0]`

Reply:
Bizarre, could you share a console log of the entire `expConfig`?

Reply:
This is the whole thing:
```
 downtimeSlasher: { slashableDowntime: 60 },
      epochRewards: { frozen: false },
      exchange: { frozen: false, minimumReports: 1 },
      exchangeEUR: { frozen: false, minimumReports: 1 },
      goldToken: { frozen: false },
      governance:
       { dequeueFrequency: 30,
         queueExpiry: 1000,
         approvalStageDuration: 100,
         referendumStageDuration: 100,
         executionStageDuration: 100,
         minDeposit: 1,
         concurrentProposals: 5 },
      governanceApproverMultiSig:
       { signatories: [ '0x5409ed021d9299bf6814279a6a1411a7e866a631' ],
         numRequiredConfirmations: 1 },
      grandaMento:
       { approver: '0x0000000000000000000000000000000000000000',
         spread: 0.01,
         vetoPeriodSeconds: 10800 },
      oracles: { reportExpiry: 300 },
      reserve:
       { initialBalance: 100000000,
         otherAddresses: [ '0x91c987bf62D25945dB517BDAa840A6c661374402' ] },
      reserveSpenderMultiSig:
       { signatories:
          [ '0x5409ed021d9299bf6814279a6a1411a7e866a631',
            '0x4404ac8bd8F9618D27Ad2f1485AA1B2cFD82482D' ],
         numRequiredConfirmations: 2 },
      stableToken:
       { goldPrice: 1,
         initialBalances: { addresses: [Array], values: [Array] },
         oracles:
          [ '0x5409ED021D9299bf6814279A6A1411A7e866A631',
            '0xE36Ea790bc9d7AB70C55260C66D52b1eca985f84',
            '0x06cEf8E666768cC40Cc78CF93d9611019dDcB628',
            '0x7457d5E02197480Db681D3fdF256c7acA21bDc12' ],
         frozen: false },
      stableTokenEUR:
       { goldPrice: 1,
         initialBalances: { addresses: [Array], values: [Array] },
         oracles:
          [ '0x5409ED021D9299bf6814279A6A1411A7e866A631',
            '0xE36Ea790bc9d7AB70C55260C66D52b1eca985f84',
            '0x06cEf8E666768cC40Cc78CF93d9611019dDcB628',
            '0x7457d5E02197480Db681D3fdF256c7acA21bDc12' ],
         frozen: false },
      validators: { commissionUpdateDelay: 3 } }
     ```